File: examples/drivers/python/basic.py
import requests
import json

BASE_URL = "http://localhost:8080"


def main():
    print("=== PrimusDB Python Example ===")

    # Check health
    try:
        resp = requests.get(f"{BASE_URL}/health", timeout=5)
        resp.raise_for_status()
        print(f"Health: {resp.json()}")
    except requests.RequestException as e:
        print(f"Error connecting to PrimusDB: {e}")
        return

    # Get version
    try:
        resp = requests.get(f"{BASE_URL}/version", timeout=5)
        resp.raise_for_status()
        print(f"Version: {resp.json()}")
    except requests.RequestException as e:
        print(f"Error fetching version: {e}")

    # Create a record
    record = {
        "collection": "users",
        "data": {
            "name": "Alice",
            "email": "alice@example.com",
        },
    }
    try:
        resp = requests.post(f"{BASE_URL}/records", json=record, timeout=5)
        resp.raise_for_status()
        print(f"Created record: {resp.json()}")
    except requests.RequestException as e:
        print(f"Error creating record: {e}")

    # Query records
    try:
        resp = requests.get(f"{BASE_URL}/records/users", timeout=5)
        resp.raise_for_status()
        print(f"Records: {resp.json()}")
    except requests.RequestException as e:
        print(f"Error querying records: {e}")


if __name__ == "__main__":
    main()

File: examples/drivers/rust/basic.rs
use std::time::Duration;

use reqwest::Client;
use serde_json::{json, Value};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let base_url = "http://localhost:8080";
    let client = Client::builder()
        .timeout(Duration::from_secs(5))
        .build()?;

    // Check health
    println!("=== PrimusDB Rust Example ===");

    match client.get(format!("{}/health", base_url)).send().await {
        Ok(resp) => {
            let health: Value = resp.json().await?;
            println!("Health: {}", health);
        }
        Err(e) => {
            eprintln!("Error connecting to PrimusDB: {}", e);
            std::process::exit(1);
        }
    }

    // Get version
    match client.get(format!("{}/version", base_url)).send().await {
        Ok(resp) => {
            let version: Value = resp.json().await?;
            println!("Version: {}", version);
        }
        Err(e) => {
            eprintln!("Error fetching version: {}", e);
        }
    }

    // Create a record
    let record = json!({
        "collection": "users",
        "data": {
            "name": "Alice",
            "email": "alice@example.com"
        }
    });

    match client
        .post(format!("{}/records", base_url))
        .json(&record)
        .send()
        .await
    {
        Ok(resp) => {
            let created: Value = resp.json().await?;
            println!("Created record: {}", created);
        }
        Err(e) => {
            eprintln!("Error creating record: {}", e);
        }
    }

    // Query records
    match client
        .get(format!("{}/records/users", base_url))
        .send()
        .await
    {
        Ok(resp) => {
            let records: Value = resp.json().await?;
            println!("Records: {}", records);
        }
        Err(e) => {
            eprintln!("Error querying records: {}", e);
        }
    }

    Ok(())
}
